credentials/jwt.go

```diff
@@ -0,0 +1,54 @@
+package credentials
+
+import (
+	"github.com/dgrijalva/jwt-go"
+	"io"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type jwtProvider struct {
+	creds JwtCredentials
+}
+
+type claims struct {
+	jwt.StandardClaims
+}
+
+func (provider *jwtProvider) Retrieve() (io.Reader, error) {
+	expirationTime := provider.GetAppropriateExpirationTime()
+	tokenString, err := provider.BuildClaimsToken(expirationTime, provider.creds.URL, provider.creds.ClientId, provider.creds.ClientUsername)
+	if err != nil {
+		return nil, err
```

Comment:
`suggestion`

I have started using error wrapping to add more information and it has really helped in debugging.